LandSandBoat/server
# [Interaction/BCNM] Apocalypse Nigh Questline Conversion | Apocalypse Nigh BCNM Cleanup

**_I affirm:_**

## What does this pull request do?
For Apocalypse Nigh BCNM:
+ Fixes spawn points for all BCNM arenas for Eald'narche and Kam'lanaut with rough approximations. This was done as each arena had Kam'lanaut slightly out of place and Eald'narche in Narnia. Both mobs should be centered in front of the player now in each arena.
+ Expands Eald'narche's functionality to include post-teleport spell casting and saving TP between warps based on videos.
+ Removed standback on Eald'narche to better represent their behavior after warps. Eald'narche should cast a spell then immediately go to close distance regardless of HP.
+ Expands the distance Eald'narche may warp to beyond 1.5' based on videos.
+ Removed the teleport mobMods from Eald'narche as this is now handled in the lua due to needing to conserve TP.

For Apocalypse Nigh Story:
+ Converts the Apocalypse Nigh questline to interaction format.
+ Simplified reward events and distribution to be more reliable and easier to read.
+ Added additional interactions at specific states in the questline provided by @KnowOne134 who has been co-authored.

Future TODOs:
+ Modify the teleport code to allow for an additional parameter that allows for conserving tp. (Would require both C++ and Lua work beyond the scope of a single fight. Unsure if other mobs using teleports should also benefit from this the same way.)

## Steps to test these changes
+ Walked through each step of the questline from Shadows of the Departed until Apocalypse Nigh rewards were given.
+ Ensured the correct reward is given based on the option chosen during the event. 
+ Tested item lockout on qm1 in Empyreal Paradox.
+ Tested that the event with qm1 is triggered when post-reward and the player has no earring. 
+ Tested that the player once reset is able to complete the appropriate storylines again.
+ Tested Kam'lanaut's lua to ensure accuracy and reliability. No issues found.
+ Tested Eald'narche's lua to ensure accuracy and reliability. This was tested for over 60 minutes in total after the final changes were complete over the course of multiple instances. No issues found as Eald'narche teleports without an issue and will cast as expected upon the end of the teleport.
